### PR TITLE
HADOOP-17004. ABFS: Improve the ABFS driver documentation

### PR DESCRIPTION
**Pull request by bilaharith**
ABFS: Improve the ABFS driver documentation.
There is no code change. So the tests has not been ran for this.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  20m 27s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 35s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  35m  3s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  14m  7s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 32s |  The patch does not generate ASF License warnings.  |
|  |   |  52m 55s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1970 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux 141c52c5c3a6 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / e069a06 |
| Max. process+thread count | 459 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-azure/src/site/markdown/abfs.md [hunk 1]
```diff
@@ -665,6 +665,79 @@ performance issues.
 Config `fs.azure.enable.check.access` needs to be set true to enable
  the AzureBlobFileSystem.access().
 
+### <a name="authconfigoptions"></a> Auth Options
+`fs.azure.account.key`: To set the account access key. Access keys can be used to authenticate the requests made from the ABFS driver to the Azure storage account.
```
Comment (snvijaya) [inline hadoop-tools/hadoop-azure/src/site/markdown/abfs.md:669]: account key config is already explained with an example in the above shared key section. Please cross check with the details above and if needed more comments can be added in respective place.

Comment (snvijaya) [reply hadoop-tools/hadoop-azure/src/site/markdown/abfs.md:669]: Comment holds for other configs too, please check.

File: hadoop-tools/hadoop-azure/src/site/markdown/abfs.md [hunk 2]
```diff
@@ -665,6 +665,79 @@ performance issues.
...
+
+`fs.azure.account.oauth2.user.name`: To set username when UserPasswordTokenProvider is used.
+
+`fs.azure.account.oauth2.user.password`: To set password when UserPasswordTokenProvider is used.
+
+`fs.azure.account.oauth2.refresh.token`: To set OAuth refreshtoken when RefreshTokenBasedTokenProvider is used.
+
+`fs.azure.account.oauth2.refresh.token.endpoint`: To set OAuth refresh token end point when RefreshTokenBasedTokenProvider is used.
+
+### <a name="featureconfigoptions"></a> Feature Options
```
Comment (snvijaya) [inline hadoop-tools/hadoop-azure/src/site/markdown/abfs.md:705]: Maybe provide more context to the feature and then list the relevant configs.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 14s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  23m 28s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  39m 58s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  16m 11s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |  60m 39s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1970 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux 09595dae32df 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 60fa153 |
| Max. process+thread count | 308 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-azure/src/site/markdown/abfs.md [hunk 3]
```diff
@@ -539,6 +563,20 @@ token when its `getAccessToken()` method is invoked.
 The declared class must implement `org.apache.hadoop.fs.azurebfs.extensions.CustomTokenProviderAdaptee`
 and optionally `org.apache.hadoop.fs.azurebfs.extensions.BoundDTExtension`.
 
+#### <a name="delegationtokensupportconfigoptions"></a> Delegation token support
+
+Delegation token support can be achieved by making the following config true
+ `fs.azure.enable.delegation.token` and specifying the config
+  CustomDelegationTokenManager to be used with the config `fs.azure.delegation.token.provider.type`.
```
Comment (snvijaya) [inline hadoop-tools/hadoop-azure/src/site/markdown/abfs.md:570]: indentation.

File: hadoop-tools/hadoop-azure/src/site/markdown/abfs.md (hunk 1 above)
Comment (bilaharith) [reply hadoop-tools/hadoop-azure/src/site/markdown/abfs.md]: Done

File: hadoop-tools/hadoop-azure/src/site/markdown/abfs.md (hunk 2 above)
Comment (bilaharith) [reply hadoop-tools/hadoop-azure/src/site/markdown/abfs.md]: Done

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  23m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  21m 32s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 30s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  37m 29s |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  37m 48s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 24s |  the patch passed  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 51 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 2 line(s) with tabs.  |
| +1 :green_heart: |  shadedclient  |  15m  9s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 28s |  The patch does not generate ASF License warnings.  |
|  |   |  79m 12s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1970 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux 38ae625b03f3 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 30ef8d0 |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/5/artifact/out/whitespace-eol.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/5/artifact/out/whitespace-tabs.txt |
| Max. process+thread count | 309 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/5/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  22m 38s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  18m 54s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 35s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  33m 26s |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  33m 47s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 28s |  the patch passed  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 2 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 2 line(s) with tabs.  |
| +1 :green_heart: |  shadedclient  |  13m 47s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |  73m  2s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1970 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux 1694f31054dd 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 62c26b91fd0 |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/6/artifact/out/whitespace-eol.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/6/artifact/out/whitespace-tabs.txt |
| Max. process+thread count | 414 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/6/console |
| versions | git=2.17.1 maven=3.6.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  22m 30s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 32s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  38m 42s |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  39m  1s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 27s |  the patch passed  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 2 line(s) with tabs.  |
| +1 :green_heart: |  shadedclient  |  15m 29s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 27s |  The patch does not generate ASF License warnings.  |
|  |   |  57m 26s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1970 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux 732bfa6ad94e 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 18d7dfbf355 |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/7/artifact/out/whitespace-tabs.txt |
| Max. process+thread count | 344 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/7/console |
| versions | git=2.17.1 maven=3.6.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  23m 47s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 32s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  40m 26s |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  40m 46s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 29s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  16m 23s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 29s |  The patch does not generate ASF License warnings.  |
|  |   |  60m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1970 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux ce9b8aff42fd 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 936bf09c374 |
| Max. process+thread count | 315 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/8/console |
| versions | git=2.17.1 maven=3.6.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  22m 44s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  38m 56s |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  39m 15s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  16m  8s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 29s |  The patch does not generate ASF License warnings.  |
|  |   |  59m 39s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1970 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux f0ed69b70c94 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d60496e6c66 |
| Max. process+thread count | 310 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1970/9/console |
| versions | git=2.17.1 maven=3.6.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**